src/cargo/util/toml_mut/manifest.rs

```diff
@@ -409,6 +411,53 @@ impl LocalManifest {
                 }
             }
         }
+
+        // continue only if the dep is fully removed
+        if status != DependencyStatus::None {
+            return;
+        }
+
+        // clean up profile sections
+        self.get_table_mut(table_path)
+        if let Some(toml_edit::Item::Table(profile_table)) = self.data.get_mut("profile") {
+            profile_table.set_implicit(true);
+            for (_, item) in profile_table.iter_mut() {
+                if let toml_edit::Item::Table(profile_table) = item {
+                    profile_table.set_implicit(true);
+                    if let Some(toml_edit::Item::Table(package_table)) = profile_table.get_mut("package") {
+                        package_table.set_implicit(true);
+                        package_table.remove(dep_key);
+                    }
+                }
+            }
+        }
+
+        // clean up patch sections
+        if let Some(toml_edit::Item::Table(patch_table)) = self.data.get_mut("patch") {
+            patch_table.set_implicit(true);
+            for (_, item) in patch_table.iter_mut() {
+                if let toml_edit::Item::Table(table) = item {
+                    table.set_implicit(true);
+                    for (key, item) in table.iter_mut() {
+                        if key.get() == dep_key {
+                            *item = toml_edit::Item::None;
+                        }
+                    }
+                }
+            }
+        }
+
+        // clean up replace section
+        if let Some(toml_edit::Item::Table(replace_table)) = self.data.get_mut("replace") {
+            replace_table.set_implicit(true);
+            for (key, item) in replace_table.iter_mut() {
+                if let Ok(spec) = PackageIdSpec::parse(key.get()) {
+                    if spec.name() == dep_key {
```

Comment:
I think this code is kind of gross. I want to rewrite it a bit with the help of `get_table_mut`.